### DISCUSSION TITLE
Can we use self signed certificate in Gateway?

### DISCUSSION THREAD

**Discussion by 9to1url** [narrowlink/narrowlink#39] [General]
as subject :-)

**Comment by SajjadPourali** [comment]
Not at this time; however, I have plans to implement it [#40](https://github.com/narrowlink/narrowlink/issues/40) after finishing the QUIC implementation https://github.com/narrowlink/narrowlink/pull/38

**Reply by 9to1url** [reply]
Great :-)
that means will have UDP at transport and this can benefit real-time apps like streaming or calls or launch videos much faster.

**Reply by SajjadPourali** [reply]
True, the main drawback of Narrowlink in comparison to other tools was the lack of P2P support. However, there is a workaround to implement QUIC/HTTP3 for the gateway connections.

The self-signed certificate will most probably be available in the next release, but the next release will require more testing on P2P communications, especially in countries with internet restrictions like China

**Reply by 9to1url** [reply]
What is the progress for QUIC and UDP transport? should I create a new discussion for that?

**Reply by SajjadPourali** [reply]
It is already done and available on the nightly version, but it needs more time for testing.
You can try it with `--p2p` arg.

-----
P.S. : 0.2.x config is not compatible with 0.1.x configs and client token must be changed.

**Reply by 9to1url** [reply]
how to install the nightly version? :-)

**Reply by 9to1url** [reply]
can I still go thru Gateway? I want that setup as well

**Reply by SajjadPourali** [reply]
The GitHub repository refers to the latest nightly version. You can also download the binaries from https://github.com/narrowlink/narrowlink/releases/tag/Nightly. For configuration, please refer to the configuration samples for each component instead of relying on the documentation that refers to version `0.1.x`


The p2p functionality is dynamic in Narrowlink. When you create a tunnel, it attempts to create a peer-to-peer (p2p) channel. However, in the meantime, it continues to use the normal channel. If the p2p channel creation is successful, it switches to p2p mode; otherwise, it continues to use the normal mode.

**Reply by SajjadPourali** [reply]
> can I still go thru Gateway? I want that setup as well


Certainly, the p2p connection is only enabled when you use `--p2p`.

**Reply by 9to1url** [reply]
Great, let me build local, that more easy to git pull :-)
